src/ros_vision_interaction/src/vision-interaction/tests/test_vision_project_delegator.py

```diff
@@ -312,3 +317,226 @@ def test_new_day_update(vision_project_delegator, statedb):
                 "num of days since last goal setting": 0,
             }
             check_database_values(statedb, **expected_values)
+
+
+def test_reading_evaluator(vision_project_delegator, statedb):
+    test_periods = [
+        [
+            "2021-03-15 00:00:00",
+            "2021-03-16 00:00:00",
+            "2021-03-17 00:00:00",
+            "2021-03-18 00:00:00",
+            "2021-03-19 00:00:00",
+            "2021-03-20 00:00:00",
+            "2021-03-21 00:00:00",
+            "2021-03-22 00:00:00",
+            "2021-03-23 00:00:00",
+        ],
+        # month changes during test period
+        [
+            "2021-02-25 00:00:00",
+            "2021-02-26 00:00:00",
+            "2021-02-27 00:00:00",
+            "2021-02-28 00:00:00",
+            "2021-03-01 00:00:00",
+            "2021-03-02 00:00:00",
+            "2021-03-03 00:00:00",
+            "2021-03-04 00:00:00",
+            "2021-03-05 00:00:00",
+        ],
+        # year changes during test period
+        [
+            "2020-12-29 00:00:00",
+            "2020-12-30 00:00:00",
+            "2020-12-31 00:00:00",
+            "2021-01-01 00:00:00",
+            "2021-01-02 00:00:00",
+            "2021-01-03 00:00:00",
+            "2021-01-04 00:00:00",
+            "2021-01-05 00:00:00",
+            "2021-01-06 00:00:00",
+        ]
+    ]
+    for period in test_periods:
+        vision_project_delegator._reset_database()
+        # day 1
+        with freezegun.freeze_time(period[0]):
+            vision_project_delegator.update()
+            first_interaction_datetime = datetime.datetime(2021, 3, 15, 2, 0, 0, 0)
+            statedb.set("first interaction datetime", first_interaction_datetime)
+            statedb.set("last interaction datetime", first_interaction_datetime)
+            statedb.set("is done eval today", True)
+            statedb.set("current eval score", 1)
+        # day 2
+        with freezegun.freeze_time(period[1]):
+            vision_project_delegator.update()
+            expected_values = {
+                "num of days since last eval": 0,
```

Comment:
This might be less painful with constants. One thing that'd be nice is that they will autocomplete. You can also namespace them through files and classes, etc.